sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala

```diff
@@ -74,8 +82,11 @@ private[connector] trait TestV2SessionCatalogBase[T <: Table] extends Delegating
   }
 
   def clearTables(): Unit = {
-    assert(!tables.isEmpty, "Tables were empty, maybe didn't use the session catalog code path?")
+    assert(
+      tableCreated.get,
```

Comment:
Why doesn't `!tables.isEmpty` work?

Reply:
We have tests like the following:
```scala
  test("saveAsTable: Append mode should not fail if the table already exists " +
    "and a same-name temp view exist") {
    withTable("same_name") {
      withTempView("same_name") {
```
Before, the table was not cleaned up correctly, so `!tables.isEmpty` was fine. But now that tables are cleaned up with this PR, we need a different way to check if this code path was executed.